### PR TITLE
ci: add various scan & enforce them

### LINKED ISSUE #20: Add security analysis tools
- [x] Snyk
 - [x] Snyk Code
 - [x] Code QL
 - [ ] All reporting into GitHub Security Advisories

### PR DESCRIPTION
**Pull request by louis-bompart**
- Add a GHA composite action that runs Snyk and Snyk Code

- Add aforementioned action to the CI job. In the case of the CI, the goal is only to report (so this is why there's a `continue-on-error`). The enforcing is done using the checks that this generates. You can see an example in the coveo/cli: https://github.com/coveo/cli/pull/794/checks?check_run_id=6525846881. The boon of this method is that it allows PR to come in if they do not worsen the project (so if the project were to have a vuln detected, it will not grind the CI to a halt).

- Add the action to the CD, but this time, it'll crash if there's unaddressed vulnerability.
  - For Snyk dependency analysis: vuln has to be fixed or addressed using Snyk UI.
  - For Snyk Code and CodeQL, use the [GitHub Code Scanning Alert UI](https://github.com/coveo/semantic-monorepo-tools/security/code-scanning).
  
----

fixes #20

### CODE REVIEW

**Comment by louis-bompart** [comment]
Needs #36 to resync package-lock.json, which will allow snyk dependency scan to work properly

File: scripts/get-scan-results.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const owner = "coveo";
+const repo = "semantic-monorepo-tools";
+const octokit = getOctokit(process.env.GITHUB_TOKEN);
+
+const getSnykCodeAlerts = () => {
+  return octokit.rest.codeScanning.listAlertsForRepo({
+    owner,
+    repo,
+    ref: "master",
```
Comment (GermainBergeron) [inline scripts/get-scan-results.js:11]: Shouldn't that be main?
```suggestion
    ref: "main",
```

**Review by GermainBergeron — APPROVED** [review]
I'm really not that knowledgeable in Github actions, but it seems like we repeat a lot of code?

File: scripts/get-scan-results.js (hunk 1 above)
Comment (louis-bompart) [reply scripts/get-scan-results.js:11]: good catch!

**Comment by louis-bompart** [comment]
> I'm really not that knowledgeable in Github actions, but it seems like we repeat a lot of code?

Oh yeah. :feelsgood: . There's the concept of composite actions, that I wanted to use for that... But they do not support the `continue-on-error` option. So I was forced to repeat.

Tho I'll check again it seems it's been added very very recently. OneMoreTry ™️

**Comment by louis-bompart** [comment]
image
e.g. of the end results. SnykCode, Snyk Open Source are enforced (i.e. cannot merge if red)

**Pull request merged**